IlmBase/Imath/ImathMatrix.h

```diff
@@ -871,12 +1151,563 @@ Vec4<S>                    operator * (const Vec4<S> &v, const Matrix44<T> &m);
 // Typedefs for convenience
 //-------------------------
 
+typedef Matrix22 <float>  M22f;
+typedef Matrix22 <double> M22d;
 typedef Matrix33 <float>  M33f;
 typedef Matrix33 <double> M33d;
 typedef Matrix44 <float>  M44f;
 typedef Matrix44 <double> M44d;
 
 
+//---------------------------
+// Implementation of Matrix22
+//---------------------------
+
+template <class T>
+inline T *
+Matrix22<T>::operator [] (int i)
+{
+    return x[i];
+}
+
+template <class T>
+inline const T *
+Matrix22<T>::operator [] (int i) const
+{
+    return x[i];
+}
+
+template <class T>
+inline
+Matrix22<T>::Matrix22 ()
+{
+    memset (x, 0, sizeof (x));
+    x[0][0] = 1;
+    x[1][1] = 1;
+}
+
+template <class T>
+inline
+Matrix22<T>::Matrix22 (T a)
+{
+    x[0][0] = a;
+    x[0][1] = a;
+    x[1][0] = a;
+    x[1][1] = a;
+}
+
+template <class T>
+inline
+Matrix22<T>::Matrix22 (const T a[2][2]) 
+{
+    memcpy (x, a, sizeof (x));
+}
+
+template <class T>
+inline
+Matrix22<T>::Matrix22 (T a, T b, T c, T d)
+{
+    x[0][0] = a;
+    x[0][1] = b;
+    x[1][0] = c;
+    x[1][1] = d;
+}
+
+template <class T>
+inline
+Matrix22<T>::Matrix22 (const Matrix22 &v)
+{
+    memcpy (x, v.x, sizeof (x));
+}
+
+template <class T>
+template <class S>
+inline
+Matrix22<T>::Matrix22 (const Matrix22<S> &v)
+{
+    x[0][0] = T (v.x[0][0]);
+    x[0][1] = T (v.x[0][1]);
+    x[1][0] = T (v.x[1][0]);
+    x[1][1] = T (v.x[1][1]);
+}
+
+template <class T>
+inline const Matrix22<T> &
+Matrix22<T>::operator = (const Matrix22 &v)
+{
+    memcpy (x, v.x, sizeof (x));
+    return *this;
+}
+
+template <class T>
+inline const Matrix22<T> &
+Matrix22<T>::operator = (T a)
+{
+    x[0][0] = a;
+    x[0][1] = a;
+    x[1][0] = a;
+    x[1][1] = a;
+    return *this;
+}
+
+template <class T>
+inline T *
+Matrix22<T>::getValue ()
+{
+    return (T *) &x[0][0];
+}
+
+template <class T>
+inline const T *
+Matrix22<T>::getValue () const
+{
+    return (const T *) &x[0][0];
+}
+
+template <class T>
+template <class S>
+inline void
+Matrix22<T>::getValue (Matrix22<S> &v) const
+{
+    if (isSameType<S,T>::value)
+    {
+        memcpy (v.x, x, sizeof (x));
+    }
+    else
+    {
+        v.x[0][0] = x[0][0];
+        v.x[0][1] = x[0][1];
+        v.x[1][0] = x[1][0];
+        v.x[1][1] = x[1][1];
+    }
+}
+
+template <class T>
+template <class S>
+inline Matrix22<T> &
+Matrix22<T>::setValue (const Matrix22<S> &v)
+{
+    if (isSameType<S,T>::value)
+    {
+        memcpy (x, v.x, sizeof (x));
+    }
+    else
+    {
+        x[0][0] = v.x[0][0];
```

Comment:
This comment would apply to the whole file, not just Matrix22, but there should be a possibly narrowing cast here. I think something like
```cpp
x[0][0] = static_cast<T>(v.x[0][0]);
...
```
would suffice

Reply:
addressing it in a separate forum is fine :)